### PR TITLE
more single exam table related simplifications

### PR DESCRIPTION
**Pull request by esotrope**
1. adds enums for subject and assessment type
2. removes now unused assessment repo
3. removes now unneeded logic in group exam service

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Assessment.java [hunk 1]
```diff
@@ -13,10 +13,10 @@
     private long id;
     private String name;
     private long gradeId;
-    private long typeId;
-    private long subjectId;
+    private AssessmentType type;
+    private Subject subject;
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Assessment.java]: @bgsavage - these enums will break the frontend

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Subject.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package org.opentestsystem.rdw.reporting.exam;
+
+/**
+ * Represents an assessment subject
+ */
```
Comment (agorina) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Subject.java]: Could you please add a comment that this has to match to the entry in the DB?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Subject.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public enum Subject {
+
+    MATH(1),
+    ELA(2);
+
+    private final long value;
```
Comment (agorina) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Subject.java]: Not a big deal, but ```int``` should be big enough.

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Represents an assessment type
+ */
+public enum AssessmentType {
+
+    ICA(1, true),
+    IAB(2, false),
+    SUMMATIVE(3, true);
+
```
Comment (agorina) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: As we have talked, this should be moved into Common. I will do it at some point.

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Assessment.java (hunk 1 above)
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Assessment.java]: actually scratch that I changed the assessment mapper to fix it

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    private final long value;
+    private final boolean claims;
+
+    AssessmentType(final long value, final boolean claims) {
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: I know there was some push-back in an earlier review about putting business logic in enums. Not quite sure how i feel about this, but i understand the argument: whether an assessment has claims or not probably doesn't belong in the enum constraint that simply defines the types. OTOH, the enum could be used as the business-logic vehicle.

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Subject.java (hunk 3 above)
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Subject.java]: Hmm, may not be a big deal, but i strongly request it be changed to `int`

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Assessment.java (hunk 1 above)
Comment (bgsavage) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/Assessment.java]: great thanks!

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java (hunk 5 above)
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: I have no issue with moving it to the service layer - might end up creating repeated logic though. Also, if we adopt the "business logic not belonging in enums" idea fully I would suggest we also move the static "fromValue" mapping methods to the service layer as well. Thoughts?

Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: Good point.

Comment (agorina) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: Just FYI: I am planning on moving this and Subject into Common. Will wait until all open items are resolved and do it then.

Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: If we think about it the from value mapping logic is kind of something that should be entirely encapsulated into the repo layer I think

Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: The only counterargument i have is that it is very important that these values are consistent between the various repository implementations. So they should be shared. I guess we could have a shared utility class that does that mapping. Or, we could embed the necessary info in the enum  ;)

Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/exam/AssessmentType.java]: Did everything just come full circle? x D
If we have to ship a util with every enum that seems a little bit overkill i think.
Should we just leave it the way it is for now?

**Pull request merged**